application/tests/unit/common/models/EmailQueueTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+namespace tests\unit\common\models;
+
+use common\models\EmailQueue;
+use common\models\EventLog;
+use common\models\User;
+use tests\helpers\EmailUtils;
+use tests\unit\fixtures\common\models\EmailQueueFixture;
+use tests\unit\fixtures\common\models\EventLogFixture;
+use tests\unit\fixtures\common\models\UserFixture;
+use yii\codeception\DbTestCase;
+
+/**
+ * Class EmailQueueTest
+ * @package tests\unit\common\models
+ * @method User users($key)
+ * @method EventLog eventLogs($key)
+ */
+class EmailQueueTest extends DbTestCase
+{
+    
+    public function fixtures()
+    {
+        return [
+            'email_queues' => EmailQueueFixture::className(),
+            'event_log' => EventLogFixture::className(),
+            'users' => UserFixture::className(),
+        ];
+    }
+
+    public function testDefaults()
+    {
+        $emailQueue = new EmailQueue();
+        $emailQueue->to_address = 'test@test.com';
+        $emailQueue->subject = 'test subject';
+        $emailQueue->text_body = 'test body';
+        $emailQueue->save();
+
+        $this->assertEquals(0, $emailQueue->attempts_count);
+        $this->assertNotNull($emailQueue->id);
+        $this->assertNotNull($emailQueue->created);
+    }
+
+    public function testInvalidEventLogUserId()
+    {
+        $emailQueue = new EmailQueue();
+        $emailQueue->to_address = 'test@test.com';
+        $emailQueue->subject = 'test subject';
+        $emailQueue->text_body = 'test body';
+        $emailQueue->event_log_user_id = 44;
+
+        $this->assertFalse($emailQueue->save());
+        $this->assertNotNull($emailQueue->getFirstError('event_log_user_id'));
+    }
+
+    public function testInvalidAddresses()
+    {
+        $emailQueue = new EmailQueue();
+        $emailQueue->to_address = 'invalid-address';
+        $emailQueue->cc_address = 'invalid-address';
+        $emailQueue->subject = 'test subject';
+        $emailQueue->text_body = 'test body';
+
+        $this->assertFalse($emailQueue->save());
+        $this->assertNotNull($emailQueue->getFirstError('to_address'));
+        $this->assertNotNull($emailQueue->getFirstError('cc_address'));
+    }
+
+    public function testSendOrQueueSent()
+    {
+        EmailUtils::removeEmailFiles();
+        $data = [
+            'toAddress' => 'test@test.com',
+            'subject' => 'test subject - 1461087894',
+            'text_body' => 'test body',
+        ];
+
+        EmailQueue::sendOrQueue(
+            $data['toAddress'],
+            $data['subject'],
+            $data['text_body']
+        );
+
+        $this->assertTrue(EmailUtils::hasEmailFileBeenCreated($data['subject']));
```

Comment:
If you really wanted to be thorough, you could do this same assertion (except as an `assertFalse`) just before the call to `EmailQueue::sendOrQueue(...)` to make sure the system is in the expected state before you call the function.